conans/model/options.py

```diff
@@ -18,14 +18,24 @@ def __init__(self, name, value, possible_values=None):
         self._name = name
         self._value = value  # Value None = not defined
         # possible_values only possible origin is recipes
-        if possible_values is None or possible_values == "ANY":
+        if possible_values is None:
             self._possible_values = None
         else:
+            # This can contain "ANY"
             self._possible_values = [str(v) if v is not None else None for v in possible_values]
 
+    def dumps(self, scope=None):
+        if self._value is None:
+            return None
+        if scope:
+            return "%s:%s=%s" % (scope, self._name, self._value)
+        else:
+            return "%s=%s" % (self._name, self._value)
+
     def copy_conaninfo_option(self):
         # To generate a copy without validation, for package_id info.options value
-        return _PackageOption(self._name, self._value)
+        assert self._possible_values is not None  # this should always come from recipe, with []
+        return _PackageOption(self._name, self._value, self._possible_values + ["ANY"])
```

Comment:
I don't get this, so for the package id computation, is every option value is ok?

Reply:
This is before the package_id computation, but the ``package_id()`` method can change values with ``self.info.options.shared = "whatever"``. So the ``ANY`` option must be added to the possibilities if we don't want it to raise an exception.

That still needs a value defined, and will raise if the value is ``None`` (not defined)